python/ray/serve/_private/deployment_state.py

```diff
@@ -2100,7 +2100,7 @@ def _check_and_update_replicas(self):
                         "application": self.app_name,
                     },
                 )
-                self._stop_replica(replica, graceful_stop=False)
+                self._stop_replica(replica, graceful_stop=True)
```

Comment:
There should be a relevant unit test that fails after this change and should be updated.

If not, it needs to be added :)

Reply:
> There should be a relevant unit test that fails after this change and should be updated

Agreed, I'm waiting on CI to see what fails/passes after this change.

Reply:
Tests passed, so something is missing :)